pkg/controllers/cache/cache.go

```diff
@@ -152,8 +153,20 @@ func (jc *jobCache) Update(obj *v1alpha1.Job) error {
 	if !found {
 		return fmt.Errorf("failed to find job <%v>", key)
 	}
-	job.Job = obj
 
+	var oldResourceversion, newResourceversion uint64
+	var err error
+	if oldResourceversion, err = strconv.ParseUint(job.Job.ResourceVersion, 10, 64); err != nil {
+		return fmt.Errorf("failed to parase job <%v> resource version <%s>", key, job.Job.ResourceVersion)
+	}
+
+	if newResourceversion, err = strconv.ParseUint(obj.ResourceVersion, 10, 64); err != nil {
+		return fmt.Errorf("failed to parase job <%v> resource version <%s>", key, obj.ResourceVersion)
+	}
+	if newResourceversion < oldResourceversion {
+		return fmt.Errorf("job <%v> has too old resource version: %d (%d)", key, newResourceversion, oldResourceversion)
```

Comment:
If the newResourceversion is less than odResourceversion, the cache keep the same as it was, giving a warning log and returning nil seems more reasonable. 
fox example, in the killJob func, if cc.cache.Update(newJob) return error in this case, the podgroup will not be cleaned in system.

Reply:
If just log a warning log and return nil will not trigger request rerun, Actually,  `processNextReq` in `job_controller` will check `st.Execute(action)` return result, if has error, it will re-queue this request and next will get the latest job from cache.
killJob func also follow the above rule.